### ISSUE TITLE
Rules for Running a NodeConf

### ISSUE THREAD

**Issue by mikeal** [nodeconf/collective#5]
Here's a draft of the initial terms I'd like to put forward.
# Any Event

The NodeConf Collective is open to supporting and collaborating with any event in the JavaScript community regardless of its name (doesn't need to be called "NodeConf"). However, there are some basic rules all events need to abide by.
- Must adopt a Code of Conduct.
- Must be "community oriented."
- If selling sponsorships a 10% discount must be offered to Corporate Members of the Node Foundation. However, the collective will send your sponsorship materials to the events contact at each company at your request.
- If selling tickets a 10% discount must be offered to Individual Members of the Node Foundation. However, the collective will send information about your event to the entire membership at your request.
- If there is a "vendor area" or "swag area" the Node Foundation should be extended, without charge, the option of using those areas with the same treatment as your lowest sponsorship level that includes those considerations. _In practice this will be rare as the foundation has limited resources in sending representatives to events._
## Community Oriented

The NodeConf Collective exists to help bring the developer community together and to strengthen the bonds that hold us together. Many people volunteer their time to make these events happen and to help support new events and organizers and we do so because we love this community and want to see it grow and flourish. It is important that events in the collective share these goals and principals.
### Neutrality

It is important to understand that the financial liabilities in running an event can be quite large, especially for larger multi-day events. It isn't reasonable to always expect individuals to carry this responsibility. We should encourage companies, especially small ones, to take these kinds of risks and help us build a better community. At the same time, we should give them tools and constraints that help them remain incentivized without sacrificing their neutrality. Events that do not maintain neutrality have a much harder time remaining consistent year over year. Sponsors do not want to sponsor an event they feel second class at and similar sentiments can be found in speakers and attendees.

In order to stay community oriented it is important that events remain "neutral" with regard to their treatment of speakers, attendees and sponsors. However, it is common for a company to extend its own resources (employee time, finance, credit & risk) in order to get an event off the ground. When a company is backing an event this way the best way to remain neutral is to simply grant a sponsorship at a particular level and to not extend more consideration than that. Speakers and attendees from that company should not be extended special privileges.

It is also common for a company, when backing an event, to staff it. In this case the company should be listed in the "about" or "staff" section of any materials as if they were any other contributor volunteering their time to staff the event.
# So, you wanna call it a NodeConf :)
## Single Day Event
- The name of the event must not reference the region or country it is in (EU, US, West Coast, Nordic, etc). Instead the name _must_ reference the city or other attribute unique to that city or locality.
### How to Start
- We have excellent documentation that is being continually improved to help you in [organizing your event](https://github.com/nodeconf/collective/blob/master/GUIDE.md).
- Create an Issue in this repository letting us know the place and time window you're looking to do your event in and we'll create a repository for you and help you get up and running. There is no approval process for single day events.
## Multi-Day Event
- The name of the event should not extend past the country it is in without approval from the collective. _Note that some events are grandfathered in because they existed prior to this policy. However, those events do not have a lock on the larger region their name represents._
### How to start
- Log an Issue here with the event information and some information about previous events you've run. If you haven't run an event before we strongly suggest you try a single day event first, it's much easier and the financial liability are considerably less.

**Comment by mikeal** [comment]
Time for feedback!

Pinging JSConf people for input: @cramforce, @voodootikigod, @janl, @slexaxton, @serrynaimo

Pinging some other people that may not be watching this repo: @cianomaidin @thefoxis

**Comment by iancrowther** [comment]
If selling sponsorships a 10% discount must be offered to Corporate Members of the Node Foundation. However, the collective will send your sponsorship materials to the events contact at each company at your request.

can you list them for clarity

**Comment by iancrowther** [comment]
If selling tickets a 10% discount must be offered to Individual Members of the Node Foundation. However, the collective will send information about your event to the entire membership at your request.

why not make it free? they do enough already hey!

**Comment by iancrowther** [comment]
yep agree with the rest

**Comment by mikeal** [comment]
> can you list them for clarity

The members list is still growing, but we can link to it.

**Comment by mikeal** [comment]
> why not make it free? they do enough already hey!

Individual Membership in the foundation can be aquired by anybody. It's free if you're a member of a GitHub org in the foundation or $99 a year.

**Comment by janl** [comment]
> We should encourage companies, especially small ones, to take these kinds of risks

I’m not sure how to read this. On my first try it reads like we should encourage small companies to take huge financial risks that might ruin them, but I’m sure that’s not what you meant :D

The rest looks fine.

**Comment by voodootikigod** [comment]
I have severe reservations about this mainly from my experiences in attempting to run a conference in python outside of the PSF. This feels very extortionistic and not community-oriented to levy these sorts of demands with zero tangible return on investment for a struggling or even established conference. Questions I see that require addressing:
- Does the 10% only convey if the collective actually produces any value to the event? If not, essentially its 10% for someone to possibly solicit the sponsorship materials -- there are no guarantees or timeliness definitions within the process that the material was ever sent, just "good faith effort".
- Does the 10% slide with the ticket price or is it a most favored nation clause (aka 10% off best ticket price)? This open ended and would need definition. Also it needs to be made clear that if the event is sold out, there are not "special reserved" tickets that need to be provided as a consideration. In short the responsibility to procure tickets at the discounted rate rests with the collective members not with the conference.
- Without any statistics (yes chicken and egg, but in this case the conference has establishment, the collective needs to prove its value IMHO) this could very easily be a massive detriment to a conference. Since there is no volume cap on the 10% tickets, what is to stop company A, B, C in the collective from buying up all the tickets and not providing sponsorship? This could be catastrophic to the budget of an event. 
- This clause scares me: `If there is a "vendor area" or "swag area" the Node Foundation should be extended, without charge, the option of using those areas with the same treatment as your lowest sponsorship level that includes those considerations. In practice this will be rare as the foundation has limited resources in sending representatives to events.` As for most people the lowest sponsorship level could just not include swag or vendor area (very common) and will be a point of frustration. There needs to be a cap on volume and a requirement of agreement between organizer and vendor before things are allowed. While the comment is made that "it is rare" that is only because the foundation is just starting up it would be foolish to think resources will be limited in perpetuity. This could become a very much abused item that would cause problems for organizers.

The main problem I have with this is the "collective" feels like a special membership club where the members get benefits and the benefit providers (conferences) get little more than a promise to attempt to help out. I would contend the agreement needs to be much more balanced. Even in the PSF, the PSF will provide some base level of sponsorship if membership does not come to the table that way the conference has a base level commitment that entering into this will be worthwhile. Outside of that, my other core concern is that the foundation is open and membership is public, so why would I need to provide 10% discount to access a public list? Is this suggesting that conference leveraging the collective get some privileged access? If so, 1) thats not community oriented by nature and 2) its not what I thought we wanted for the goals of the foundation.

**Comment by thefoxis** [comment]
I'm signing up underneath everything that @voodootikigod has written above. 

Moreover this needs more definition and enforcement: 

> Must adopt a Code of Conduct.

I can easily see people doing the old [copy/paste game](https://modelviewculture.com/pieces/a-code-of-conduct-is-not-enough) and nothing more. This doesn't perpetuate nor enforce a safe community. Organisers should not only have CoC but also know how to act on it and react to possible situations of harassment. We should actively work on promoting diversity and inclusivity by having diversity tickets, sponsorships, balanced line-ups and attendee ratios, care about accessibility and more. This is hard work, but not impossible to do. I've seen a few people follow what I've and others done and with willingness and knowledge there's action. 

I'd way rather have a how-to, beginners guide to pursuing that goal, as I described [here](https://blog.andyet.com/2015/07/10/the-road-to-diversity-at-andyetconf). There are plenty other materials too. **We need to set an example for people to follow**. This is not something we should compromise on.

**Comment by voodootikigod** [comment]
Not to start echo volleyball, but @thefoxis is spot on. Also if the collective/foundation wants to provide something that will be helpful for conferences and a benefit, it should be a central place for reporting and maintaining code of conduct concerns. In this way a list of bad actors could be maintained across all events where right now it is entirely decentralized which is extremely problematic. By the collective/foundation handling this it can ensure the proper legal and financial components are handled uniformly across each event. Furthermore it would be great to have a non-partisan code of conduct reporting system that would facilitate/allow for the reporting of issues against organizers as well. Right now, each code of conduct basically requires reports to go to the organizers or someone representing the organizer team. This can get very tricky very quickly if something arises due to the organizers, even if inadvertently. Ideally if the community really cares about inclusivity and creating a safe environment, then the community needs to take the responsibility, ownership, and risk of ensuring and maintaining that, not the necessarily just the conferences/events (of which can lead to the aforementioned copy/paste CYA game). 

tl;dr if we can centralized the code of conduct maintenance and and reporting/handling, that is a benefit for the conferences and solves other derivative issues with `must adopt a code of conduct`. It then starts to make this a balanced equation and can be useful for driving inclusivity and balance throughout the node community by using the events as supporters not as fiefdoms.

**Comment by mikeal** [comment]
@voodootikigod @thefoxis 

On the subject of give/take with the foundation.

The foundation is providing a few things that I don't think are accurately conveyed:
- Holds and enforces the trademark, which is actually our only method of enforcing _any_ rules for a NodeConf. The legal cost here could be zero and it could be substantial depending on how many companies try to use the brand without following the guidelines.
- In the case of sponsorship, they are sending the sponsorship materials for a given conference to the proper contacts at all of the member companies. For people like me and @voodootikigod this isn't that big of a deal because we've built a large list of contacts over the last half decade of running events but this is a huge deal for newer organizers and events. I chose 10% off because it's the sort of discount I would expect to offer myself in this kind of situation in order to entice members to sponsor, if that is way off-base for others we can reduce to something like 5%.
- There seems to be some confusion about the ticket discount for "Individual Members." Companies are _not_ individual members of the foundation. The concern you laid out about companies using this to purchase as ton of tickets seemed to think this was the case. Individual members is for _individual developers_ who join the foundation (they get an email address, some swag discounts, and collectively elect a board seat). To become a member you pay $99 a year unless you are an existing contributor (member of one of the GitHub orgs in the foundation) in which case you can register for free. It is this list of people that would be emailed a discount code for themselves to use (once) when signing up. Again, I picked 10% because it's the sort of discount I would give to a new email list of "members" i was marketing to, we can reduce if that seems more reasonable.

For the vendor booth bit I have no idea what you want to see because I've literally never had a vendor booth area at any conference I've done. What this is designed to do is allow the foundation's marketing rep to have a small space (comparable to a very low level sponsor) if they choose to. I tried to write it in such a way that the foundation is not entitled to anything else that a sponsor would receive other than the space (for instance, they would still need to buy a ticket).

> Outside of that, my other core concern is that the foundation is open and membership is public, so why would I need to provide 10% discount to access a public list?

That's not entirely true. The list of companies is public but the contact for sponsorship is not. The individual membership list w/ emails may not be public either.

**Comment by mikeal** [comment]
@voodootikigod @thefoxis On Code of Conduct.

The reason I opted to require that they adopt a Code of Conduct rather than dictate one is two fold. The first is that if we just require a blanket CoC I think it's _less_ likely that they read it and are going to enforce it than if we require that they at least Google and copy/paste one. The second reason is that I've seen some new innovation happening lately around CoC's, iterations on the language that defines harassment and better guidelines for enforcement. Ideally we would see conferences adopting these as they get refined without being held back by having to adhere to which ever one we pick today.

> Also if the collective/foundation wants to provide something that will be helpful for conferences and a benefit, it should be a central place for reporting and maintaining code of conduct concerns. In this way a list of bad actors could be maintained across all events where right now it is entirely decentralized which is extremely problematic.

I definitely want to see this happen and it's something we're already starting to work towards. 

https://github.com/nodejs/diversity/issues/2

> Furthermore it would be great to have a non-partisan code of conduct reporting system that would facilitate/allow for the reporting of issues against organizers as well. Right now, each code of conduct basically requires reports to go to the organizers or someone representing the organizer team. This can get very tricky very quickly if something arises due to the organizers, even if inadvertently.

This sounds like something worth putting some time in to and building the tools around. Let's bring this to the diversity WG to get their input as well. Once a tool like this exists we can recommend or require its usage in the guidelines.

> I'd way rather have a how-to, beginners guide to pursuing that goal, as I described here. There are plenty other materials too. We need to set an example for people to follow. This is not something we should compromise on.

One of the goals of getting all the organizers together in this collective was to build out better documentation and howto's and lower the barrier to entry for creating new great conferences which is why the [GUIDE](https://github.com/nodeconf/collective/blob/master/GUIDE.md) is here. I don't see this as something that should _block_ the adoption of these guidelines though, this will be a living document and will include more links to this documentation as people write it.

**Comment by serrynaimo** [comment]
So... Let's see. I'm organizing a JS event and obviously would love to have Node guys involved. Now, it's very community oriented, we do enforce the JSConf code-of-conduct and already give discounts to relevant community groups, but it's generally 15% so I assume that's okay. We officially don't allow swag or give-aways from non-sponsors at the event as we sell this is a privilege for those who support the event financially. In practice, we usually looking the other way if it's open-source stuff and no obvious commercial interest behind it and hope our sponsors won't be ripping our head off about it. So that should be all good.
Now in exchange we get what exactly? We're now allowed to host Node foundation people that I would otherwise have to uninvite? oO What am I missing?

Correct me if I'm wrong, please, this is supposed to encourage and educate people on how to run an event, yet all it applies is restrictions. Maybe an incentive to do the right thing would be good.

**Comment by janl** [comment]
On re-reading, I’m with Chris and Karolina.

**Comment by voodootikigod** [comment]
@mikeal Just to make sure, my comments are not meant to cause trouble, just trying to make things better and avoid long term issues. Feedback is meant to be friendly/constructive in nature, which is why I am trying to provide as much info and where possible (if I have an idea) a positive solution.

Another thing, regarding not forcing a specific code of conduct, I understand the intention, but also know there is the chance that someone says OK I will have  code of conduct, code of conduct is "do whatever". That does abide by the rules and yet violates the intent. I think there has to be a bare minimum definition of what is a valid code of conduct.

RE: exchanges "Holds and enforces the trademark" wow, I feel like we took 100 steps backwards.... so let me get this straight, under corporate ownership this wasn't an issue, under foundation ownership its aggressively go after infringements -- something smells about that? To expand further 

```
Holds and enforces the trademark, which is actually our only method of enforcing any rules for a NodeConf.
```

This collective, as described, is beyond just formal NodeConf named events at least as I understand the ruleset to read. This is why there is a section called "So, you wanna call it a NodeConf :)" is this a correct understanding?

As a community person who helped get NodeConf and many others off the ground and sometimes stabilized, I offer the list of sponsorship people for free, without restrictions or "payback" but rather encouragement to share and share alike (so help others). That offer is and will always be open, though currently due to family issues is on hold. 

RE: Members, thanks for the clarification  -- except that you have two bullet points, 3 and 4, that provide the discount to both Corporate and Individual members -- hence my concern which is not addressed by your response. "The concern you laid out about companies using this to purchase as ton of tickets seemed to think this was the case." is in reference to bullet point 3 while your response seems to be from bullet point 4. 

RE: vendor booth/swag -- because you haven't doesn't mean others won't run into this. I would ask something to the effect of "with approval from the organizer" so that the foundation can be held to the same possible restrictions as the other sponsors (AKA can't send defamatory things or duplicate items or massive booth, etc.) The structure of the sentence seems to allow the foundation to send anything at all in the absence of a fully documented sponsorship outline. If we can add some words like "the organizers due retain the right of refusal for any vendor swag or vendor area items in line with any other sponsor". To make it explicitly clear that the Foundation has to abide by the same governance as any other sponsor.  (this is a simple one, hence why I am trying to give you as many word options -- don't take verbosity as anger/complexity).

Thanks

**Comment by mikeal** [comment]
@serrynaimo

I'm gonna need to re-write this doc because it is not explaining things well enough, there seems to be a lot of confusion.

> We're now allowed to host Node foundation people that I would otherwise have to uninvite?

Absolutely not. That's not what this is at all, and I don't even know how we would go about doing something like that.

> Now in exchange we get what exactly?

What we're building is a support system. This is going to be a lot more useful to newer organizers, or organizers looking to run a larger event than they have before, then it will established organizers like yourself. But, in order to explain some of the benefits I'll roll through this:

If you wanted JSConf.asia in you'd get: promotion of the event through twitter and on the website, etc. You'd get to send a 10% discount link to a bunch of people that may otherwise be hard to reach. You get to send a 10% off discount offer of sponsorship to a bunch of sponsorship contacts at companies you may not have otherwise had access to. You have no booth area so that isn't a thing, if the foundation sent some node.js stickers you'd put them out. That's about it.

Now, let's say that as part of Dev Week Singapore you want to have a single day NodeConf and rather than do it all yourself you want to enable a new organizer to run it. They would have the support of prior organizers, help promoting the CFP, help with budgeting, and also access to the documentation we're building (although that is public and open to everyone). Additionally you'd get all the stuff I mentioned above.

But, you should be able to run an event outside the collective and have node people involved. This shouldn't be a deterrent. To Chris' point earlier, I don't want this to turn in to the thing Chris had to deal with running a Python event outside the PSF where he essentially got yelled at by a bunch of people for just doing something without some kind of permission.

**Comment by mikeal** [comment]
@voodootikigod 

> Just to make sure, my comments are not meant to cause trouble, just trying to make things better and avoid long term issues.

This is all very helpful. So far my biggest takeaway is that all the language in the rules is terrible and miss-represents my intention and pretty much every detail :)

> Another thing, regarding not forcing a specific code of conduct, I understand the intention, but also know there is the chance that someone says OK I will have code of conduct, code of conduct is "do whatever". That does abide by the rules and yet violates the intent. I think there has to be a bare minimum definition of what is a valid code of conduct.

This is a good point, and something I hadn't thought of. I'm going to walk back my previous comment and implement a blanket CoC. I'm also now realizing that whatever shape the Diverity WG's reporting stuff takes it will probably require a particular CoC as enforcing conduct for a multitude of CoC's would be rather difficult.

> "Holds and enforces the trademark"

Just to back up: today my LLC owns the mark, similar to how your LLC owns the JSConf mark. We've both had rules up for a long time about running an event using the mark and have had to ask events to change their name when they didn't comply. For me, I'm worried about this increasing due to the commercialization of Node, especially abroad, and I don't have the financial resources to pay the legal bill here. If I don't enforce the mark we loose it, cause trademark law is painful that way.

> This collective, as described, is beyond just formal NodeConf named events at least as I understand the ruleset to read. This is why there is a section called "So, you wanna call it a NodeConf :)" is this a correct understanding?

Correct, my comment about the mark was specific to events named "NodeConf."

> Discount abuse for buying ticket lots.

I'm not seeing how this would work. So, each individual member (contributor or paying $99 per year) would have the option of buying _a single ticket_ at a 10% discount. If a company wanted to use this discount to purchase tickets in bulk they would have to sign up all of their people as members of the foundation first. For most of these events $99 is more than 10% of the ticket cost.

The goal of the discount is to get people more invested long term, becoming members of the foundation, engaged in the election process, and attending _multiple_ events per year. If successful the members list is an ideal place to promote each event because virtually everyone on it is buying a few tickets a year.

The Corporate Member discount _only_ applies to sponsorship.

**Comment by mikeal** [comment]
@voodootikigod oh, and your last comment about the vendor booth stuff makes sense. I need to alter this dramatically to make it clear that this is at the discretion of the organizer and must follow any rules they've put forward.

**Comment by voodootikigod** [comment]
@mikeal I think by making your text clearly like this:

If selling sponsorships a 10% discount on **_only sponsorships**_... 
If selling tickets a 10% discount on **_only tickets**_...

would go along way to solving the problem, I didn't even see the specificity (completely my bad) but it would be lost to others as well. make sure to call out the difference a little bigger and maybe not have them one after the other as it runs together.

Thanks for clarifying other parts.

RE: CoC use the word "minimum definition of a CoC" instead of blanket. Difference between build your own up from here and here just use this. I think we all would prefer the former.

**Comment by iancrowther** [comment]
How can I start to get the help regarding budgeting and CFP reviews? It's a bit open ended and whilst I'd be grateful for any support from the foundation I'm a) not sure what to expect and b) not sure how to solicit this help
